lib/schemaUtils.js

```diff
@@ -1293,10 +1296,72 @@ module.exports = {
 
       // stop searching for helper if valid auth scheme is found
       if (!_.isEmpty(helper)) {
+        if (_.isEmpty(authHelper)) {
+          authHelper = helper;
+        }
+        else if (helper.type === 'apikey') {
+          authHelper.extraAPIKeys = !authHelper.extraAPIKeys ? [] : authHelper.extraAPIKeys;
+          authHelper.extraAPIKeys.push(helper);
+        }
+      }
+
+      if (!_.isEmpty(authHelper) && !shouldCheckForMultipleKeys) {
+        // Already got a security schema no need to check further
         return false;
       }
     });
-    return helper;
+    return authHelper;
+  },
+
+  getExtraAPIKeyParams: function (authHelper) {
+    const extraAPIKeys = _.get(authHelper, 'extraAPIKeys', []);
+    let headers = [],
+      queryParams = [];
+
+    // Remove extraAPIKeys property
+    _.set(authHelper, 'extraAPIKeys', undefined);
+
+    extraAPIKeys.forEach((apiKey) => {
+      let keyValuePair = {
+        key: _.get(apiKey, 'apikey.0.value'),
+        value: _.get(apiKey, 'apikey.1.value')
+      };
+
+      if (_.get(apiKey, 'apikey.2.value') === 'header') {
+        headers.push(keyValuePair);
+      }
+      else {
```

Comment:
let's use `else if ( === query ) {` condition specifically to represent this better here. As API key can be mentioned inside `cookie` as well.